packages/ra-core/src/controller/create/useCreateController.ts

```diff
@@ -60,28 +58,28 @@ export const useCreateController = <
     const recordToUse = record ?? getRecordFromLocation(location) ?? undefined;
     const { onSuccess, onError, ...otherMutationOptions } = mutationOptions;
 
-    const [create, { isLoading: saving }] = useCreate(
+    const [create, { isLoading: saving }] = useCreate<RecordType>(
         resource,
         undefined,
         otherMutationOptions
     );
 
     const save = useCallback(
-        (
```

Comment:
I don't understand these changes. Do they change anything? If not, why is the new code better?

Reply:
The async keywords were added while debugging. I'll remove them. However, we now return the promise at L77, just like in `useEditController`